hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/ProcessorSupplierWithService.java

```diff
@@ -50,7 +52,10 @@ private ProcessorSupplierWithService(
 
     @Override
     public void init(@Nonnull ProcessorSupplier.Context context) {
-        this.serviceContext = serviceFactory.createContextFn().apply(context);
+        HazelcastInstanceImpl hazelcastInstance = (HazelcastInstanceImpl) context.jetInstance().getHazelcastInstance();
+        ManagedContext managedContext = hazelcastInstance.getSerializationService().getManagedContext();
+        serviceContext = serviceFactory.createContextFn().apply(context);
+        serviceContext = (C) managedContext.initialize(serviceContext);
```

Comment:
do we have any guarantee that always same type is returned? should we throw a more meaningful exception if it's not the case?